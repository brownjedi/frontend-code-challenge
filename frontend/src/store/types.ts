import { State as PokedexState } from './modules/pokedex'

export interface RootState {
  pokedex: PokedexState
}
