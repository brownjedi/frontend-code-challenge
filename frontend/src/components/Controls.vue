<template>
  <div :class="styles.container">
    <Search />
  </div>
</template>

<script lang="ts">
import { defineComponent, ref, useCssModule, watch } from 'vue'
import { usePokemonTypesQuery } from '@/composables/pokemonsTypesQuery'
import Search from './Search.vue'

export default defineComponent({
  components: { Search },
  setup() {
    const styles = useCssModule()
    const { types, loading, error } = usePokemonTypesQuery()
    const value = ref('')

    watch(value, value => console.log(value))

    return { styles, types, value, loading }
  },
})
</script>

<style lang="scss" module>
.container {
  display: flex;
  margin-bottom: 3rem;
}
</style>
