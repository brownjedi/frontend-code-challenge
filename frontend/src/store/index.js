import Vue from 'vue'
import Vuex from 'vuex'

// Load all modules.
function loadModules() {
  const context = require.context('./modules', false, /([a-z_]+)\.js$/i)

  const modules = context
    .keys()
    .map(key => ({ key, name: key.match(/([a-z_]+)\.js$/i)[1] }))
    .reduce(
      (modules, { key, name }) => ({
        ...modules,
        [name]: context(key).default,
      }),
      {}
    )

  return { context, modules }
}

const { modules } = loadModules()

Vue.use(Vuex)

export default new Vuex.Store({
  strict: process.env.NODE_ENV !== 'production',
  modules,
})
