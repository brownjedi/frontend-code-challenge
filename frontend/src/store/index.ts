import { createStore, createLogger } from 'vuex'
import { RootState } from './types'

import { store as pokedex, PokedexStore } from './modules/pokedex'

export type Store = PokedexStore<Pick<RootState, 'pokedex'>>

export const store = createStore<RootState>({
  plugins: [createLogger()],
  modules: {
    pokedex,
  },
})

export function useStore(): Store {
  return store as Store
}
