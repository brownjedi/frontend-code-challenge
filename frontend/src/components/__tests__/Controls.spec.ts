import { shallowMount } from '@vue/test-utils'
import Controls from '@/components/Controls.vue'
import { store } from '@/store'

describe('components/Controls.vue', () => {
  it('has the expected html structure', () => {
    const wrapper = shallowMount(Controls)
    expect(wrapper.element).toMatchSnapshot()
  })

  it('updates the vuex store when search', async () => {
    const wrapper = shallowMount(Controls, {
      global: {
        plugins: [store],
      },
    })

    expect(store.state.pokedex.query.search).toBe('')

    const searchString = 'pikachu'
    await wrapper.vm.onSearch(searchString)

    expect(store.state.pokedex.query.search).toBe(searchString)
  })
})
