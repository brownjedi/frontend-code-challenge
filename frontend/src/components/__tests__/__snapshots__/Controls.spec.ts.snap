// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Controls.vue has the expected html structure 1`] = `
<div
  class="container"
>
  <search-stub
    class="search"
    initialvalue=""
  />
</div>
`;
